### PR TITLE
Add address and thread sanitization

### PR DESCRIPTION
**Pull request by oleorhagen**

### CODE REVIEW

File: cmake/asan.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
+get_property(isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+
+if(isMultiConfig)
+    if(NOT "ASan" IN_LIST CMAKE_CONFIGURATION_TYPES)
+        list(APPEND CMAKE_CONFIGURATION_TYPES Asan)
+    endif()
+else()
+    set(allowedBuildTypes ASan Debug Release RelWithDebInfo MinSizeRel)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "${allowedBuildTypes}")
+endif()
```
Comment (vpodzime) [inline cmake/asan.cmake:10]: Still don't see the world where this is nicer, simpler and easier to understand than autotools...

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,6 +12,9 @@ endif (POLICY CMP0135)
 
 project(mender)
 
+include(cmake/asan.cmake)
+include(cmake/memsan.cmake)
+include(cmake/threadsan.cmake)
```
Comment (vpodzime) [inline CMakeLists.txt:17]: I guess you took these from somewhere? Would be nice to mention it in the commit message. Also, what's the license?

**Review by vpodzime — COMMENTED** [review]
Looks OK to me otherwise. CI doesn't like it, though.

File: cmake/asan.cmake (hunk 1 above)
Comment (oleorhagen) [reply cmake/asan.cmake:10]: How is this helpful?

File: CMakeLists.txt (hunk 2 above)
Comment (oleorhagen) [reply CMakeLists.txt:17]: I adapted them from `learning CMake` the book. I'll mention it in the PR :+1: 

It still feels like this should be able to meta-program a bit though, like this is bacially a function rendering this template with two variables...

Comment (oleorhagen) [reply CMakeLists.txt:16]: Simplified it quite a bit now. Turns out CMake has functions :tada:

**Review by vpodzime — COMMENTED** [review]
I think the first 3 commits can be squashed. Looks good to me otherwise.

File: cmake/helper.cmake [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+
+function (sanitizer_add_compiler_flags CONFIG SANITIZER_BUILD_FLAGS SANITIZER_SHARED_LINKER_FLAGS)
+
+  message (STATUS "Adding the flags: ${SANITIZER_BUILD_FLAGS}")
```
[the remark this answers is not in the document]
Comment (oleorhagen) [reply cmake/helper.cmake]: Yes, good point. Added :)

**Review by kacf — APPROVED** [review]
👍

**Pull request merged**